pkg/zhttpclient/client.go

```diff
@@ -0,0 +1,98 @@
+package zhttpclient
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type ZHTTPClient interface {
+	SetRetryPolicy(retryPolicy *RetryPolicy) ZHTTPClient
+	NewRequest() ZRequest
+	Do(ctx context.Context, req *http.Request) (int, []byte, error)
+}
+
+type Config struct {
```

Comment:
This struct (and properties) should be private, because otherwise it can be used once the client is initialized